python_modules/libraries/dagstermill/dagstermill/asset_factory.py

```diff
@@ -185,6 +187,9 @@ def iris_dataset():
 
     default_tags = {"notebook_path": _clean_path_for_windows(notebook_path), "kind": "ipynb"}
 
+    if safe_is_subclass(config_schema, Config):
+        config_schema = infer_schema_from_config_class(cast(Type[Config], config_schema))
+
```

Comment:
We should consider making `UserConfigSchema` accept this and then make `convert_user_facing_definition_config_schema` handle the coercion.

Reply:
Going to log this as a follow up in our doc, but leave this codepath for now.